scripts/validate_config_filenames.py

```diff
@@ -1,40 +1,58 @@
-"""Check if zone files are valid. Zone files must be uppercase."""
-for file in zone_files:
-    if file.endswith(".yaml") or file.endswith(".yml"):
-        file = file.replace(".yaml", "") or file.replace(".yml", "")
-        if file != file.upper():
-            zone_error = True
-            print(f"ERROR: {file} is not uppercase")
-if zone_error == False:
-    print("All zone filenames are valid!")
-elif zone_error == True:
-    print("There are errors in the above zone filenames!")
-
-"""Check if exchange files are valid. Exchange files must be sorted and uppercase."""
-for file in exchange_files:
-    if file.endswith(".yaml") or file.endswith(".yml"):
-        file = file.replace(".yaml", "") or file.replace(".yml", "")
-        exchange_keys = file.split("_")
-        sorted_exchange_keys = sorted(exchange_keys)
-        if file != file.upper():
-            exchange_error = True
-            print(f"ERROR: {file} is not uppercase")
-        if exchange_keys != sorted_exchange_keys:
-            exchange_error = True
-            print(f"ERROR: {file} is not sorted")
-if exchange_error == False:
-    print("All exchange filenames are valid!")
-elif exchange_error == True:
-    print("There are errors in the above exchange filenames!")
-
-
-if zone_error or exchange_error:
-    exit(1)
+
+def main():
+    zone_files = os.listdir("config/zones")
+    exchange_files = os.listdir("config/exchanges")
+    zone_error: bool = False
+    exchange_error: bool = False
+
+    print("Checking config files...")
+    print("Checking if zone files are valid...")
+
+    # Check if zone files are valid. Zone files must be uppercase.
+    for file in zone_files:
+        if file.endswith(".yaml") or file.endswith(".yml"):
+            file = file.replace(".yaml", "") or file.replace(".yml", "")
+            if file != file.upper():
+                zone_error = True
+                print(f"ERROR: {file} is not uppercase")
+    if zone_error:
+        print("There are errors in the above zone filenames!")
+    else:
+        print("All zone filenames are valid!")
+
+    # Check if exchange files are valid. Exchange files must be sorted and
+    # uppercase.
+    for file in exchange_files:
+        if file.endswith(".yaml") or file.endswith(".yml"):
+            file = file.replace(".yaml", "") or file.replace(".yml", "")
+            exchange_keys = file.split("_")
+            sorted_exchange_keys = sorted(exchange_keys)
+            if file != file.upper():
+                exchange_error = True
+                print(f"ERROR: {file} is not uppercase")
+            if exchange_keys != sorted_exchange_keys:
+                exchange_error = True
+                print(f"ERROR: {file} is not sorted")
+    if exchange_error:
+        print("There are errors in the above exchange filenames!")
+    else:
+        print("All exchange filenames are valid!")
+
+    if zone_error or exchange_error:
+        exit(1)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
I'm not really sure why this is needed, won't it add additional overhead (albeit small).

Reply:
Thanks for review.

This is intended to improve readability (mostly through consistency). Python scripts in general, including all other scripts here, usually have functions at the top level, and the only statement at the top level is `if __name__ == "__main__": main()` at the bottom.

Having a consistent style across scripts (and across code in general) makes them easier to read. Some reasons to organize code in functions and conditionally call main at the bottom are:

1. it makes it so that it's possible to import modules without creating side-effects (https://realpython.com/python-main-function/#best-practices-for-python-main-functions).
2. it's a familiar pattern for Python programs and across programming languages.

This PR also changes a couple of other unusual or un-idiomatic things:

```
    if zone_error == False:
        print('OK')
    elif zone_error == True:
        print('Errors')
```
becomes
```
    if zone_error:
        print("Errors")
    else:
        print("OK")
```

And string literals above statements (which are not file or function docstrings) become regular Python comments.

Reply:
hmm I'm not sure if changing the if statments here make sense, for one the first one is more clear as to what exactly we are checking and more type safe as the second one is only checking if zone_error is truthy and printing ok in every other case, even if it would be undefined.

There is also the order of operations to consider here, since we expect all names to be error free 99.9% of the time it makes sense to check and print that first, to avoid the secondary else or elif statment. This is into micro optimization territory but it's similar to an early return pattern that is commonly used.

Reply:
Okay let's get rid of the strict Boolean check then, not I don't really get why it would be a anti pattern in a loosely typed language but that's a whole different thing.

But I do think we should change back the order of the if else statement like the last example you provided.

Reply:
Yep, makes sense, and I see what you're saying. Now changed. I also thought maybe using has_zone_error is slightly more explicit/clear than just zone_error?

Note that this is also a convention other languages. The explanation given is just "comparing booleans with boolean literals is unnecessary, directly using the booleans is more concise".

 - https://typescript-eslint.io/rules/no-unnecessary-boolean-literal-compare/
 - https://dartcodemetrics.dev/docs/rules/common/no-boolean-literal-compare
 - https://progmiscon.org/misconceptions/Java/ComparisonWithBooleanLiteral/